text/text.go

```diff
@@ -53,22 +53,23 @@ const (
 )
 
 const (
-	Thin       Weight = 100 - 400
-	Hairline   Weight = Thin
-	ExtraLight Weight = 200 - 400
-	UltraLight Weight = ExtraLight
-	Light      Weight = 300 - 400
-	Normal     Weight = 400 - 400
-	Medium     Weight = 500 - 400
-	SemiBold   Weight = 600 - 400
-	DemiBold   Weight = SemiBold
-	Bold       Weight = 700 - 400
-	ExtraBold  Weight = 800 - 400
-	UltraBold  Weight = ExtraBold
-	Black      Weight = 900 - 400
-	Heavy      Weight = Black
-	ExtraBlack Weight = 950 - 400
-	UltraBlack Weight = ExtraBlack
+	Thin Weight = 100*iota - 300
+	ExtraLight
+	Light
+	Normal
+	Medium
+	SemiBold
+	Bold
+	ExtraBold
+	Black
+
+	Hairline   = Thin
+	UltraLight = ExtraLight
+	DemiBold   = SemiBold
+	UltraBold  = ExtraBold
+	Heavy      = Black
+	ExtraBlack = Black + 50
+	UltraBlack = ExtraBlack
```

Comment:
I think the use of `iota` actually makes this less clear, and invites errors if some new weight is inserted in the list.  I think I'd just prefer literals:

```go
Thin       Weight = -300
ExtraLight Weight = -200
Light      Weight = -100
Normal     Weight = 0
Medium     Weight = 100
SemiBold   Weight = 200
Bold       Weight = 300
ExtraBold  Weight = 400
Black      Weight = 500
```

I do like the consolidation of `Hairline` - `UltraBlack` at the bottom, I think that does make things clearer.

Reply:
Since the list is comprehensive from thin to black, I don't think a new entry will arrive, maybe to either end. Mid values (+50) could be added to the second half.

Also any decent ide will show a constant's value when you hover your mouse over it.

Reply:
I agree with @theclapp that we should not use `iota` to compute these constants. I also think we should get rid of the duplicates, and perhaps even cut down on the number of constants. In any case, ok to do in a future change.